src/app/components/team/Languages/LanguageListItem.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { graphql, commitMutation } from 'react-relay/compat';
+import { Store } from 'react-relay/classic';
+import IconButton from '@material-ui/core/IconButton';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
+import ListItemText from '@material-ui/core/ListItemText';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import Typography from '@material-ui/core/Typography';
+import IconMoreVert from '@material-ui/icons/MoreVert';
+import { FormattedGlobalMessage } from '../../MappedMessage';
+import ConfirmProceedDialog from '../../layout/ConfirmProceedDialog';
+import languagesList from '../../../languagesList';
+
+function submitLanguage({ input, onCompleted, onError }) {
+  commitMutation(Store, {
+    mutation: graphql`
+      mutation LanguageListItemUpdateTeamMutation($input: UpdateTeamInput!) {
+        updateTeam(input: $input) {
+          team {
+            id
+            get_language
+            get_languages
+          }
+        }
+      }
+    `,
+    variables: {
+      input,
+    },
+    onCompleted,
+    onError,
+  });
+}
+
+const LanguageListItem = ({
+  code,
+  isDefault,
+  languages,
+  team,
+}) => {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [defaultDialogOpen, setDefaultDialogOpen] = React.useState(false);
+  const [deleteDialogOpen, setDeleteDialogOpen] = React.useState(false);
+
+  const handleClose = () => setAnchorEl(null);
+
+  const handleMenuDelete = () => {
+    handleClose();
+    setDeleteDialogOpen(true);
+  };
+
+  const submitDelete = () => {
+    const onCompleted = () => {
+      setDeleteDialogOpen(false);
+    };
+    const onError = () => {
+      setDeleteDialogOpen(false);
+    };
+
+    submitLanguage({
+      input: {
+        id: team.id,
+        languages: JSON.stringify(languages.filter(l => l !== code)),
+      },
+      onCompleted,
+      onError,
+    });
+  };
+
+  const handleMenuMakeDefault = () => {
+    handleClose();
+    setDefaultDialogOpen(true);
+  };
+
+  const submitDefault = () => {
+    const onCompleted = () => {
```

Comment:
I'm using the `isSaving`/`CircularProgress` pattern here. As with the statuses, we don't have individual permissions for languages so it could be a bit tricky to block menu actions on optimistic data.